webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -349,28 +351,57 @@ func (v validator) validateStorageClass(ctx *context.WebhookRequestContext, vm *
 	scPath := field.NewPath("spec", "storageClass")
 	scName := vm.Spec.StorageClass
 
-	// TODO: This validation shouldn't be done in the webhook.
-
-	sc := &storagev1.StorageClass{}
-	if err := v.client.Get(ctx, client.ObjectKey{Name: scName}, sc); err != nil {
+	zones, err := topology.GetAvailabilityZones(ctx, v.client)
+	if err != nil {
 		return append(allErrs, field.Invalid(scPath, scName, fmt.Sprintf(storageClassNotFoundFmt, vm.Namespace)))
 	}
 
-	resourceQuotas := &corev1.ResourceQuotaList{}
-	if err := v.client.List(ctx, resourceQuotas, client.InNamespace(vm.Namespace)); err != nil {
-		return append(allErrs, field.Invalid(scPath, scName, err.Error()))
-	}
+	if pkgconfig.FromContext(ctx).Features.PodVMOnStretchedSupervisor && len(zones) > 1 {
+		storagePolicyQuota := &cnsstoragev1.StoragePolicyQuota{}
+		if err := v.client.Get(ctx, client.ObjectKey{
+			Namespace: vm.Namespace,
+			Name:      util.CNSStoragePolicyQuotaName(scName),
+		}, storagePolicyQuota); err != nil {
+			return append(allErrs, field.Invalid(scPath, scName, fmt.Sprintf(storageClassNotFoundFmt, vm.Namespace)))
+		}
 
-	prefix := scName + storageResourceQuotaStrPattern
-	for _, resourceQuota := range resourceQuotas.Items {
-		for resourceName := range resourceQuota.Spec.Hard {
-			if strings.HasPrefix(resourceName.String(), prefix) {
-				return nil
+		// TODO(srm09): Is this validation necessary?
```

Comment:
I don't think so, at least not yet. This is just replacing how we enforce the storage NS association, and since the new thing is per-class specific and namespace-scoped that's implied.